File: Python Codes/10_Jump_Game.py
def can_jump(nums):
    mx = 0
    for i in range(len(nums)):
        if mx < i:
            return False
        mx = max(mx, i + nums[i])
    return True

if __name__ == "__main__":
    nums = [2, 3, 1, 1, 4] 
    if can_jump(nums):
        print("You can reach the end of the array!\n true")
    else:
        print("You cannot reach the end of the array.\n false")

File: CPP Codes/10_Jump_Game.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

bool canJump(const vector<int>& nums) {
    int mx = 0;
    for (int i = 0; i < nums.size(); ++i) {
        if (mx < i) {
            return false;
        }
        mx = max(mx, i + nums[i]);
    }
    return true;
}

int main() {
    vector<int> nums = {2, 3, 1, 1, 4}; // Example input
    if (canJump(nums)) {
        cout << "You can reach the end of the array!" << endl;
    } else {
        cout << "You cannot reach the end of the array." << endl;
    }
    return 0;
}
